native/src/boot/bootimg.cpp

```diff
@@ -457,25 +457,40 @@ bool boot_img::parse_image(const uint8_t *p, format_t type) {
         }
         fprintf(stderr, "%-*s [%s]\n", PADDING, "KERNEL_FMT", fmt2name[k_fmt]);
     }
+
     if (auto size = hdr->ramdisk_size()) {
         if (hdr->is_vendor() && hdr->header_version() >= 4) {
-            // v4 vendor boot contains multiple ramdisks
-            // Do not try to mess with it for now
-            r_fmt = UNKNOWN;
+            for (int i = 0; i < hdr->vendor_ramdisk_table_entry_num(); ++i) {
+                auto &[entry, fmt] = ramdisk_table_entries.emplace_back(make_unique<vendor_ramdisk_table_entry_v4>(), UNKNOWN);
+                memcpy(entry.get(),
+                       vendor_ramdisk_table + (i * hdr->vendor_ramdisk_table_entry_size()),
+                       hdr->vendor_ramdisk_table_entry_size());
+                fmt = check_fmt_lg(ramdisk + entry->ramdisk_offset, entry->ramdisk_size);
+            }
         } else {
-            r_fmt = check_fmt_lg(ramdisk, size);
+            auto &[entry, fmt] = ramdisk_table_entries.emplace_back(make_unique<vendor_ramdisk_table_entry_v4>(), UNKNOWN);
+            entry->ramdisk_size = hdr->ramdisk_size();
+            entry->ramdisk_offset = 0;
+            entry->ramdisk_name[0] = '\0';
+            fmt = check_fmt_lg(ramdisk, hdr->ramdisk_size());
```

Comment:
same as b4

Reply:
Normal ramdisk processing is done as before and doesn't rely on v4 table entries anymore